projects/gloo/pkg/translator/translator.go

```diff
@@ -131,16 +131,15 @@ func (t *translatorInstance) Translate(
 	var endpoints []*envoy_config_endpoint_v3.ClusterLoadAssignment
 	clusters, endpoints = t.translateClusterSubsystemComponents(params, proxy, reports)
 	routeConfigs, listeners := t.translateListenerSubsystemComponents(params, proxy, proxyReport)
+
 	// run Resource Generator Plugins
 	for _, plugin := range t.pluginRegistry.GetResourceGeneratorPlugins() {
-		generatedClusters, generatedEndpoints, generatedRouteConfigs, generatedListeners, err := plugin.GeneratedResources(params, clusters, endpoints, routeConfigs, listeners)
-		if err != nil {
-			reports.AddError(proxy, err)
-		}
-		clusters = append(clusters, generatedClusters...)
-		endpoints = append(endpoints, generatedEndpoints...)
-		routeConfigs = append(routeConfigs, generatedRouteConfigs...)
-		listeners = append(listeners, generatedListeners...)
+		newClusters, newEndpoints, newRouteConfigs, newListeners := plugin.GeneratedResources(params, proxy,
```

Comment:
`GeneratedResources` has a comment about being deprecated, this (the only non-test caller) doesn't have anything like a `if (is ggv2) continue`? Should we have that? Otherwise maybe we should clarify the deprecation comment.

Reply:
There are a few other plugins that use GeneratedResources in the `gateway2` translation path, so we can't skip it. The [check here](https://github.com/solo-io/gloo/blob/ac6bd9fe60164e93af9fc5ab41ac5b7af38b664a/projects/gateway2/proxy_syncer/perclient.go#L85) prevents the HTTP tunnel plugin from double adding the clusters. 

When the other plugins are updated, we can pull what you're highlghting out and only use `UpstreamGeneratedResources` for both the gloo and gateway2 translation paths.

I will add a comment clarifying the intended approach to removing this.

Reply:
I'm adding:
```
// GeneratedResources is deprecated and being replaced by UpstreamGeneratedResources
// The plan is to move the few remaining plugins in Gloo/EE to use the new
// interface. Once that is done and the new interface is called in the Gloo translation
// path, we can remove this code.
// During this transition period, gateway2 will call both interfaces and it's translator
// has (must have) checks that prevent duplicate resources from being added.
```